lib/diffy/diff.rb

```diff
@@ -114,6 +114,29 @@ def each_chunk
       end
     end
 
+    def each_hunk
+      # Iterate over "change hunks": contigous hunks of context around a
+      # change. Only works with unified diffs:
+      # https://en.wikipedia.org/wiki/Diff_utility#Unified_format
+      if not @options[:diff] =~ /-U/
+        raise "Diffy only supports each_hunk for unified diffs."
+      end
```

Comment:
This bit is a bit of a hack I'm afraid, but I think just crashing out with an error is preferable to silently failing when the diff is of the wrong type. Open to suggestions though!